packages/components/src/menu-item/stories/index.story.tsx

```diff
@@ -0,0 +1,86 @@
+/**
+ * External dependencies
+ */
+import type { Meta, StoryFn } from '@storybook/react';
+
+/**
+ * WordPress dependencies
+ */
+import { link, more, check } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import MenuGroup from '../../menu-group';
+import MenuItem, { MenuItem as UnforwardedMenuItem } from '..';
+import Shortcut from '../../shortcut';
+
+const meta: Meta< typeof UnforwardedMenuItem > = {
+	component: UnforwardedMenuItem,
+	title: 'Components/MenuItem',
+	argTypes: {
+		children: { control: { type: null } },
+		icon: {
+			control: { type: 'select' },
+			options: [ 'check', 'link', 'more' ],
+			mapping: {
+				check,
+				link,
+				more,
+			},
+		},
+	},
+	parameters: {
+		controls: {
+			expanded: true,
+			include: [
+				'className',
+				'children',
+				'info',
+				'icon',
+				'iconPosition',
+				'isSelected',
+				'role',
+				'suffix',
+				'label',
+			],
```

Comment:
This kind of tells me that we're allowing too many `Button` props 🤔  Looking at usage in the repo, I feel like `isDestructive` is the only prop that is really appropriate in this context. Maybe we should just allow that one.

```diff
diff --git a/packages/components/src/menu-item/types.ts b/packages/components/src/menu-item/types.ts
index a9249ffa20..8178959ebf 100644
--- a/packages/components/src/menu-item/types.ts
+++ b/packages/components/src/menu-item/types.ts
@@ -8,7 +8,7 @@ import type { ReactNode } from 'react';
  */
 import type { ButtonAsButtonProps } from '../button/types';
 
-export type MenuItemProps = ButtonAsButtonProps & {
+export type MenuItemProps = Pick< ButtonAsButtonProps, 'isDestructive' > & {
 	/**
 	 * A CSS `class` to give to the container element.
 	 */

```

This will raise a TS error in `packages/components/src/tools-panel/tools-panel-header/component.tsx`, but I'd rather suppress that as a TODO for now.

```diff
diff --git a/packages/components/src/tools-panel/tools-panel-header/component.tsx b/packages/components/src/tools-panel/tools-panel-header/component.tsx
index b7c9f73656..35f27983c1 100644
--- a/packages/components/src/tools-panel/tools-panel-header/component.tsx
+++ b/packages/components/src/tools-panel/tools-panel-header/component.tsx
@@ -214,6 +214,9 @@ const ToolsPanelHeader = (
 							<MenuGroup>
 								<MenuItem
 									aria-disabled={ ! canResetAll }
+									// @ts-expect-error - TODO: If this "tertiary" style is something we really want to allow on MenuItem,
+									// we should rename it and explicitly allow it as an official API. All the other Button variants
+									// don't make sense in a MenuItem context, and should be disallowed.
 									variant={ 'tertiary' }
 									onClick={ () => {
 										if ( canResetAll ) {

```

Reply:
Good point! Do you think a new issue should be introduced for taking notes about this or `TODO` note is good enough

Reply:
Sorted